package-discrepancies.sh

```diff
@@ -0,0 +1,187 @@
+    for kind in "${KINDS[@]}"; do
+        for arch in "${ARCHES[@]}"; do
+            "${action}" "${kind}" "${arch}"
+        done
+    done
+}
+
+download_package_files() {
+    if [[ -e "${WORKDIR}/amd_container_pkgs" ]] && \
+           [[ -e "${WORKDIR}/amd_image_pkgs" ]] && \
+           [[ -e "${WORKDIR}/arm_container_pkgs" ]] && \
+           [[ -e "${WORKDIR}/arm_image_pkgs" ]]; then
+        return
+    fi
+    for_each_kind_arch download_action
+}
+
+drop_src_action() {
+    local kind="${1}"
+    local arch="${2}"
+
+    local file="${WORKDIR}/${arch}_${kind}_pkgs"
+    local file_dsrc="${file}_dsrc"
+
+    sed -e 's/::.*//' "${file}" >"${file_dsrc}"
+}
+
+drop_src() {
+    if [[ -e "${WORKDIR}/amd_image_pkgs_dsrc" ]] && \
+           [[ -e "${WORKDIR}/amd_container_pkgs_dsrc" ]] && \
+           [[ -e "${WORKDIR}/arm_image_pkgs_dsrc" ]] && \
+           [[ -e "${WORKDIR}/arm_container_pkgs_dsrc" ]]; then
+        return
+    fi
+    download_package_files
+    for_each_kind_arch drop_src_action
+}
+
+merge_pkgs() {
+    if [[ -e "${WORKDIR}/amd_pkgs" ]] && \
+           [[ -e "${WORKDIR}/arm_pkgs" ]]; then
+        return
+    fi
+    drop_src
+
+    local arch
+    for arch in "${ARCHES[@]}"; do
+        LC_ALL=C sort --unique "${WORKDIR}/${arch}_"{image,container}'_pkgs_dsrc' >"${WORKDIR}/${arch}_pkgs"
+    done
+}
+
+get_pkg_names() {
+    if [[ -e "${WORKDIR}/amd_pkg_names" ]] && \
+           [[ -e "${WORKDIR}/arm_pkg_names" ]]; then
+        return
+    fi
+    merge_pkgs
+    local arch
+    for arch in "${ARCHES[@]}"; do
+        sed -e 's/-[0-9]\+\(\.[0-9]\+\)*.*//' "${WORKDIR}/${arch}_pkgs" >"${WORKDIR}/${arch}_pkg_names"
+    done
+}
+
+get_common_pkg_names() {
+    if [[ -e "${WORKDIR}/common_pkg_names" ]]; then
+        return
+    fi
+    get_pkg_names
+    LC_ALL=C comm -1 -2 "${WORKDIR}/"{amd,arm}'_pkg_names' >"${WORKDIR}/common_pkg_names"
+}
+
+# lines in the input stream must be sorted
+#
+# passed parameters must be sorted
+filter_pkgs() {
+    local pkg_name
+    local line
+    while [[ "${#}" -gt 0 ]] && read -r line; do
+        pkg_name="${1}"
+        if [[ "${line}" == "${pkg_name}-"* ]]; then
+            printf '%s\n' "${line}"
+            shift
+        fi
+    done
+}
+
+filter_common_pkgs() {
+    if [[ -e "${WORKDIR}/amd_filtered_pkgs" ]] && \
+           [[ -e "${WORKDIR}/arm_filtered_pkgs" ]]; then
+        return
+    fi
+    get_common_pkg_names
+    local line
+    local -a all_common_pkgs
+    while read -r line; do
+        all_common_pkgs+=("${line}")
+    done <"${WORKDIR}/common_pkg_names"
+    merge_pkgs
+    local arch
+    for arch in "${ARCHES[@]}"; do
+        filter_pkgs "${all_common_pkgs[@]}" <"${WORKDIR}/${arch}_pkgs" >"${WORKDIR}/${arch}_filtered_pkgs"
+    done
+}
+
+get_diff() {
+    if [[ -e "${WORKDIR}/diff_output" ]]; then
+        return
+    fi
+    filter_common_pkgs
+    { diff "${WORKDIR}/"{amd,arm}'_filtered_pkgs' || :; } >"${WORKDIR}/diff_output"
```

Comment:
```suggestion
    { sdiff -d -s "${WORKDIR}/"{amd,arm}'_filtered_pkgs' || :; } >"${WORKDIR}/diff_output"
```
I think this output is easier to read but I agree that using a standard diff output is maybe best by default. Yet then I suggest to at least add `-u` for a unified diff so that it can be passed to `ydiff -s` or similar.